### PR TITLE
Material system and Kernel system documentation

### PR DESCRIPTION
**Pull request by aeslaughter**
(refs #12049)

@lindsayad Can you add some content to the newly created AD page in this PR?

### CODE REVIEW

**Comment by lindsayad** [comment]
Ok I added some AD content!

File: framework/doc/content/source/interfaces/Coupleable.md [hunk 1]
```diff
@@ -8,8 +8,13 @@ The following table summarizes the methods and kinds of values they provide:
 coupledValue | Values of a coupled variable in q-points
 coupledGradient | Gradients of a coupled variable in q-points
 coupledSecond | Second derivatives of a coupled variable in q-points
+adCoupledValue | Values of a coupled variable in q-points with automatic differentiation info
+adCoupledGradient | Gradients of a coupled variable in q-points with automatic differentiation info
+adCoupledSecond | Second derivatives of a coupled variable in q-points with automatic differentiation info
 coupledNodalValue | Values of a coupled variable at nodes.
+adCupledNodalValue | Values of a coupled (vector) variable at nodes with automatic differentiation info
```
Comment (aeslaughter) [inline framework/doc/content/source/interfaces/Coupleable.md:15]:
```suggestion
adCoupledNodalValue | Values of a coupled (vector) variable at nodes with automatic differentiation info
```

File: framework/doc/content/syntax/Materials/index.md [hunk 2]
```diff
@@ -1 +1,259 @@
-!template load file=stubs/moose_system.md.template name=Materials syntax=/Materials
\ No newline at end of file
+# Materials System
+
+The material system is the primary mechanism for defining spatially varying properties. The system
+allows properties to be defined in single object (a `Material`) and shared among the many other
```
Comment (cticenhour) [inline framework/doc/content/syntax/Materials/index.md:4]:
```suggestion
allows properties to be defined in a single object (a `Material`) and shared among the many other
```

File: framework/doc/content/syntax/Materials/index.md [hunk 3]
```diff
@@ -1 +1,259 @@
...
+get methods. This method initializes the `_diffusivity` member variable to reference the
+desired value of the property as computed by the material object.
+
+!listing ex08_materials/src/kernels/ExampleDiffusion.C line=_diffusivity(get
+
+The name used in the get method, "diffusivity", in this case is not arbitrary. This name corresponds
+with the name used to declare the property in the material object.
+
+!alert note title=The declare/get calls must correspond
+If a material properties is declared for automatic differentiation (AD) using `declareADProperty`
```
Comment (cticenhour) [inline framework/doc/content/syntax/Materials/index.md:105]:
```suggestion
If a material property is declared for automatic differentiation (AD) using `declareADProperty`
```

File: framework/doc/content/syntax/Materials/index.md [hunk 4]
```diff
@@ -1 +1,259 @@
...
+then it +must+ be consumed with the `getADMaterialProperty`. The same is true for non-automatic
+differentiation; properties declared with `declareProperty` +must+ be consumed with the
+`getMaterialProperty` method.
+
+
+## Property Names id=property-names
+
+When creating a Material object and declaring the properties that shall be computed it is often
```
Comment (cticenhour) [inline framework/doc/content/syntax/Materials/index.md:113]:
```suggestion
When creating a Material object and declaring the properties that shall be computed, it is often
```

File: framework/doc/content/syntax/Materials/index.md [hunk 5]
```diff
@@ -1 +1,259 @@
...
+desirable to allow for the property name to be changed via the input file. This may be accomplished
+by by adding an input parameter for assigning the name. For example, considering the example above
```
Comment (cticenhour) [inline framework/doc/content/syntax/Materials/index.md:115]:
```suggestion
by adding an input parameter for assigning the name. For example, considering the example above
```

File: framework/doc/content/syntax/Materials/index.md [hunk 6]
```diff
@@ -1 +1,259 @@
...
+On the consumer side, the get method will now be required to use the name "not_diffusivity" to
+retrieve the property. Consuming objects can also use the same procedure to allow for custom
+property names by adding a parameter and using the parameter name in the get method in the same
+fashion.
+
+
+## Default Material Properties
+
+The [#material-name] input parameter also provides the ability to set default values for scalar
+(`Real`) properties. For example, in the above example the input file can use number or
```
Comment (cticenhour) [inline framework/doc/content/syntax/Materials/index.md:151]:
```suggestion
(`Real`) properties. In the above example, the input file can use number or
```
Too much use of "example"

**Review by cticenhour — CHANGES_REQUESTED** [review]
Only a few suggested changed related to grammar - everything else looks good!

**Review by cticenhour — APPROVED** [review]

**Comment by aeslaughter** [comment]
The failure (https://civet.inl.gov/job/518299/) is not related; this PR only changed comments and markdown files.

**Pull request merged**